src/client/libds3/bucket.c

```diff
@@ -7,31 +7,343 @@
 #include "ds3_internal.h"
 
 int
-ds3_bucket_list(daos_size_t *nbuck, struct ds3_bucket_info *buf, ds3_t *ds3, daos_event_t *ev)
+ds3_bucket_list(daos_size_t *nbuck, struct ds3_bucket_info *buf, char *marker, bool *is_truncated,
+		ds3_t *ds3, daos_event_t *ev)
 {
-	return 0;
+	if (ds3 == NULL || nbuck == NULL || buf == NULL || marker == NULL)
+		return -EINVAL;
+
+	int                         rc = 0;
```

Comment:
Is there a style guide or a linter that I could use that will enfoce these rules and give me an easy to read result?

Reply:
usually when you push a PR, a checkpatch program is run and tells you all the issues. however nothing is run here since you are not part of the github org. let me see what we can do here.